lib/src/config_handler.dart

```diff
@@ -29,7 +31,7 @@ class ConfigHandler {
   /// Get a string representation of the current date in the following format
   /// yyyy-MM-dd (2023-01-09)
   static String get dateStamp {
-    return DateFormat('yyyy-MM-dd').format(DateTime.now());
+    return DateFormat('yyyy-MM-dd').format(clock.now());
```

Comment:
This was initially how I did this but realized that this would required having to pass the clock from the top level `Analytics` class to every class it initializes.

Instead I read that `package:clock` avoids this for testing by using the `withClock(Clock clock, T Function() callback)` method. This allows us to pass a fake clock as the first positional parameter and it will essentially do what you have outlined but through the use of zones. I referred to the article below

https://iiro.dev/controlling-time-with-package-clock/

Reply:
I'll leave this one up to you, but I'll point out that you're direct injecting everything else, and using `withClock()` is essentially the opposite direction of https://github.com/flutter/flutter/issues/47161